arch-design/multicloud/BucketManagement.md

```diff
@@ -0,0 +1,369 @@
+
+`Valid Values`: private | public-read | public-read-write | authenticated-read
+
+`x-amz-bucket-object-lock-enabled`: Specifies whether you want S3 Object Lock to be enabled for the new bucket.
+
+`x-amz-grant-full-control`: Allows grantee the read, write, read ACP, and write ACP permissions on the bucket.
+
+`x-amz-grant-read`: Allows grantee to list the objects in the bucket.
+
+`x-amz-grant-read-acp`: Allows grantee to read the bucket ACL.
+
+`x-amz-grant-write`: Allows grantee to create, overwrite, and delete any object in the bucket.
+
+`x-amz-grant-write-acp`: Allows grantee to write the ACL for the applicable bucket.
+
+##### API:
+***
+PUT / HTTP/1.1/BucketName
+***
+##### Request Body
+The request accepts the following data in XML format.
+
+`CreateBucketConfiguration`: Root level tag for the CreateBucketConfiguration parameters. Required: Yes
+
+`LocationConstraint`: Specifies the Region where the bucket will be created. If you don't specify a Region, the bucket is created in the US East (N. Virginia) Region (us-east-1).
+
+Type: String
+
+Valid Values: af-south-1 | ap-east-1 | ap-northeast-1 | ap-northeast-2 | ap-northeast-3 | ap-south-1 | ap-southeast-1 | ap-southeast-2 | ca-central-1 | cn-north-1 | cn-northwest-1 | EU | eu-central-1 | eu-north-1 | eu-south-1 | eu-west-1 | eu-west-2 | eu-west-3 | me-south-1 | sa-east-1 | us-east-2 | us-gov-east-1 | us-gov-west-1 | us-west-1 | us-west-2
+
+Required: No
+
+##### Response body
+
+`return code`: 200
+
+`location`: specifies the Region where the bucket will be created. If you are creating a bucket on the US East (N. Virginia) Region (us-east-1), you do not need to specify the location.
+
+***
+DELETE / HTTP/1.1/Bucket
+***
+
+URI Request Parameters:
+
+The request uses the following URI parameters.
+
+`Bucket`: Specifies the bucket being deleted.
+
+Required: Yes
+
+`x-amz-expected-bucket-owner`: The account id of the expected bucket owner. If the bucket is owned by a different account, the request will fail with an HTTP 403 (Access Denied) error.
+
+##### Request body:
+No body
+
+##### Response:
+return code 204
+
+#### Azure:
+Azure cloud bucket creation:
+
+##### Rules for Bucket(known as container in azure) naming:
+
+* A container name must be a valid DNS name, conforming to the following naming rules:
+
+* Container names must start with a letter or number, and can contain only letters, numbers, and the dash (-) character.
+
+* Every dash (-) character must be immediately preceded and followed by a letter or number; consecutive dashes are not permitted in container names.
+
+* All letters in a container name must be lowercase.
+* Container names must be from 3 through 63 characters long.
+
+##### API:
+
+***
+ ##### PUT https://myaccount.blob.core.windows.net/mycontainer?restype=container
+***
+
+##### Request headers:
+
+* Authorization     Required.
+
+* Date or x-ms-date     Required.  (Can be automated or manual)
+
+* x-ms-version     Required for all authorized requests.
```

Comment:
how do we plan to keep these in sync with each cloud back end, when they change ?

Reply:
Right. It;s tricky to maintain. I have noted down the issues. But for now, we have kept default version for each cloud backends.